src/oauth/endpoints/userinfo.ts

```diff
@@ -0,0 +1,33 @@
+import { Hono } from "hono";
+import { type Variables, scopeRequired, tokenRequired } from "../middleware";
+
+const app = new Hono<{ Variables: Variables }>();
+
+app.on(["GET", "POST"], "/", tokenRequired, scopeRequired(["profile"]), (c) => {
+  const accountOwner = c.get("token").accountOwner;
+
+  if (!accountOwner) {
+    return c.json(
+      {
+        error: "This method requires an authenticated user",
+      },
+      401,
+    );
+  }
+
+  const defaultAvatarUrl = new URL(
+    "/image/avatars/original/missing.png",
+    c.req.url,
+  ).href;
+
+  return c.json({
+    iss: new URL("/", c.req.url).href,
+    sub: accountOwner.account.iri,
```

Comment:
Should I actually be asking `federation` here for these? or is it safe to trust the database here?

Reply:
Since Hollo assumes and requires users not to change their domain names once after they created their first account, I believe it's fine to trust the database here.